### PR TITLE
additional logging around dirty pages

### PR DESCRIPTION
**Pull request by tmakatos**
Signed-off-by: Thanos Makatos <thanos.makatos@nutanix.com>

### CODE REVIEW

File: lib/dma.c [hunk 1]
```diff
@@ -542,7 +544,27 @@ dma_controller_dirty_page_logging_stop(dma_controller_t *dma)
...
     dma->dirty_pgsize = 0;
+
+    vfu_log(dma->vfu_ctx, LOG_DEBUG, "dirty pages: stopped logging");
+}
+
+
+#ifdef DEBUG
+static void
+log_dirty_bitmap(vfu_ctx_t *vfu_ctx, dma_memory_region_t *region,
+                   char *bitmap, size_t size)
```
Comment (jlevon) [inline lib/dma.c:555]: indentation?

File: lib/dma.c [hunk 2]
```diff
@@ -542,7 +544,27 @@ dma_controller_dirty_page_logging_stop(dma_controller_t *dma)
...
+{
+    size_t i;
+    size_t count;
+    for (i = 0, count = 0; i < size; i++) {
+        count += __builtin_popcount(bitmap[i]);
+    }
+    vfu_log(vfu_ctx, LOG_DEBUG, "dirty pages: get %#lx-%#lx, %zu dirty pages",
+            (unsigned long)region->info.iova.iov_base,
```
Comment (jlevon) [inline lib/dma.c:563]: could we use %p and iov_end ?

File: lib/libvfio-user.c [hunk 3]
```diff
@@ -942,7 +942,12 @@ handle_dirty_pages_get(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
             msg->out_size = 0;
             return ERROR_INT(ret);
         }
+    } else {
+        vfu_log(vfu_ctx, LOG_ERR,
+                "dirty pages: get %#lx-%#lx: buffer too small (%u < %lu)",
+                range_in->iova, range_in->size, dirty_pages_in->argsz, argsz);
```
Comment (jlevon) [inline lib/libvfio-user.c:948]: iova + size no?

File: lib/dma.c (hunk 1 above)
Comment (tmakatos) [reply lib/dma.c:555]: right

File: lib/libvfio-user.c (hunk 3 above)
Comment (tmakatos) [reply lib/libvfio-user.c:948]: right

File: lib/dma.c (hunk 2 above)
Comment (tmakatos) [reply lib/dma.c]: sure

**Review by jlevon — APPROVED** [review]
ta

**Pull request merged**